internal/auth/issuer/pam/pam_issuer.go

```diff
@@ -428,14 +449,25 @@ func (s *PAMOIDCProvider) Authorize(ctx context.Context, req *pamapi.AuthAuthori
 		}
 	}
 	if !validRedirect {
-		s.log.Warnf("Authorize: invalid redirect URI - %s", req.RedirectUri)
-		return nil, errors.New(ErrorInvalidRequest)
+		var allowedURIs string
+		var allowedURIsForLog string
+		if s.config != nil && len(s.config.RedirectURIs) > 0 {
+			// For logs: show full list (server-side only)
+			allowedURIsForLog = fmt.Sprintf(" Allowed redirect URIs: %v", s.config.RedirectURIs)
+			// For client error: show count only to avoid revealing internal configuration
+			allowedURIs = fmt.Sprintf(" %d redirect URI(s) configured", len(s.config.RedirectURIs))
+		} else {
+			allowedURIs = " No redirect URIs configured"
+			allowedURIsForLog = allowedURIs
+		}
```

Comment:
could we simplify this a bit with a single var, whenever I am in a function or method need to declare 2 vars I take a couple steps back :)

```go
var allowed string

if s.config != nil && len(s.config.RedirectURIs) > 0 {
	allowed = fmt.Sprintf("%d redirect URI(s) configured", len(s.config.RedirectURIs))
} else {
	allowed = "no redirect URIs configured"
}
...